### ISSUE TITLE
Invalid syntax error when creating Jep instance

### ISSUE THREAD

**Issue by git4sun** [ninia/jep#167]
- OS Platform, Distribution, and Version: Mac, Mojave, 10.14, also tried on 10.13
- Python Distribution and Version: 3.6
- Java Distribution and Version: tried both 1.8 & 10.0
- Jep Version: 3.8.2
- Python packages used (e.g. numpy, pandas, tensorflow): numpy

Can't get jep working in a project. Always throw this exception. I did find another issue and a few other threads about not finding library path. None of the solution works so far. I have copied the .jar and .so files everywhere. But still throws the exception. 

I also installed jep in pip. It was installed successfully. But can't even run `Jep jep=new Jep();`

    jep.JepException: <class 'SyntaxError'>: ('invalid syntax', ('<string>', 1, 7, 'Jep jep=new Jep();'))

Any idea?

**Comment by bsteffensmeier** [comment]
When you are troubleshooting environment issues, the first step is to run the jep script that is installed when you install jep. This is normally installed in the bin directory of your python installation, running it from the command line should result in a Python prompt nearly identical to a standard Python prompt but it is actually a Java process running all the commands through jep. Running this script will verify that jep is installed correctly. If this script works then you can look in the script, it is just a shell script, and try to create an environment for your project that matches the environment the script creates.

> Exception in thread "main" java.lang.UnsatisfiedLinkError: no jep in java.library.path

This error typically indicates that libjep cannot be loaded by JNI. There are several ways you can solve this:

1. The most straightforward solution is to set the "java.library.path" system property to point to the directory that contains libjep.jnilib. This is usually done by changing the java command used to run your project and adding "-Djava.library.path=" followed by the path to the directory that contains libjep.jnilib
2. If you cannot change the java command running your process but you can change the environment then you can try setting the PYTHON_HOME environment variable. To find the correct value for this setting I recommend running python normally and using the value of `sys.prefix`.
3. Instead of setting PYTHON_HOME in the environment you can set it from your Java code using [PyConfig.setPythonHome()](http://ninia.github.io/jep/javadoc/3.8/jep/PyConfig.html#setPythonHome-java.lang.String-)and then [MainInterpreter.setInitParams()](http://ninia.github.io/jep/javadoc/3.8/jep/MainInterpreter.html#setInitParams-jep.PyConfig-). This must be done before using Jep.
4. In the past it has been common to use the LD_LIBRARY_PATH environmental variable and you can try this, however in recent versions of OS X they have started ignoring this variable unless you reconfigure certain OS level settings so I would not recommend this anymore for MacOS

> I have copied the .jar and .so files everywhere.

You can copy the jar wherever it is convenient for your project, I would not recommend copying the .so, since python is expecting it at a particular location, also on MacOS java typically loads libraries with the .jnilib or .dylib extension and not .so, so I copying the .so would have no affect on this particular error.

> jep.JepException: <class 'SyntaxError'>: ('invalid syntax', ('<string>', 1, 7, 'Jep jep=new Jep();'))

This exception is very different then the exception in your title. This seems to indicate you successfully loaded libjep but it is having trouble executing some python. Jep executes some python in the constructor so it is likely that is the source of the problem. If you could include more of the stack trace including the line numbers that are failing then I could look into this more.

**Comment by SoulSen** [comment]
Having this same problem, where is `libjep.jnilib` even found?

**Comment by ndjensen** [comment]
It will be in the site-packages/jep folder that your Python uses.  It varies in location based on the operating system.  When you do python setup.py install the output should tell you where the jep files were installed.

If all else fails, you should be able to find it on a Unix (macOS or Linux) system by typing `locate libjep` at the command line.

**Comment by ndjensen** [comment]
If it's still a SyntaxError, we really need the stacktrace to figure out what is going on.  The Jep constructor is calling into Python to configure some things, but without the stacktrace we can't determine which line is causing the issue and therefore cannot diagnose this.

**Comment by bsteffensmeier** [comment]
> jep.JepException: <class 'SyntaxError'>: ('invalid syntax', ('<string>', 1, 7, 'Jep jep=new Jep();'))

The syntax error is from Python and includes the Python line being executed. It looks like you are trying to instantiate a Jep instance with Java code inside Python. You should use python syntax when executing statements and you shouldn't create a Jep instance from python.

**Comment by ndjensen** [comment]
Closing due to lack of activity.
